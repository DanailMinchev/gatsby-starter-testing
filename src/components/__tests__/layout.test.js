import React from "react"
import { render, screen } from "@testing-library/react"
import { useStaticQuery } from "gatsby" // mocked

import Layout, { PureLayout } from "../layout"

describe(`Layout`, () => {
  beforeEach(() => {
    useStaticQuery.mockImplementationOnce(() => ({
      site: {
        siteMetadata: {
          title: "Gatsby Testing Starter",
        },
      },
    }))
  })

  it(`renders a header`, () => {
    const { container } = render(
      <Layout>
        <main>
          <h1>Hello World</h1>
        </main>
      </Layout>
    )

    expect(container.querySelector(`header`)).toBeInTheDocument()
  })

  it(`renders children`, () => {
    const text = `Hello World`
    render(
      <Layout>
        <main>
          <h1>{text}</h1>
        </main>
      </Layout>
    )

    const child = screen.getByText(text)
    expect(child).toBeInTheDocument()
  })

  describe(`Snapshots`, () => {
    it(`matches snapshot`, () => {
      const dateMock = new Date("2020-06-22T14:45:46.773Z")
      const dateSpy = jest
        .spyOn(global.Date.prototype, "getFullYear")
        .mockImplementationOnce(() => dateMock.getFullYear())

      const { container } = render(
        <Layout>
          <main>
            <h1>Hello World</h1>
          </main>
        </Layout>
      )

      expect(dateSpy).toHaveBeenCalled()
      expect(container).toMatchSnapshot()

      dateSpy.mockRestore()
    })
  })
})

describe(`PureLayout`, () => {
  const data = {
    site: {
      siteMetadata: {
        title: "Gatsby Testing Starter",
      },
    },
  }

  it(`renders a header`, () => {
    const { container } = render(
      <PureLayout data={data}>
        <main>
          <h1>Hello World</h1>
        </main>
      </PureLayout>
    )

    expect(container.querySelector(`header`)).toBeInTheDocument()
  })

  it(`renders children`, () => {
    const text = `Hello World`
    render(
      <PureLayout data={data}>
        <main>
          <h1>{text}</h1>
        </main>
      </PureLayout>
    )

    const child = screen.getByText(text)
    expect(child).toBeInTheDocument()
  })

  describe(`Snapshots`, () => {
    it(`matches snapshot`, () => {
      const dateMock = new Date("2020-06-22T14:45:46.773Z")
      const dateSpy = jest
        .spyOn(global.Date.prototype, "getFullYear")
        .mockImplementationOnce(() => dateMock.getFullYear())

      const { container } = render(
        <PureLayout data={data}>
          <main>
            <h1>Hello World</h1>
          </main>
        </PureLayout>
      )

      expect(dateSpy).toHaveBeenCalled()
      expect(container).toMatchSnapshot()

      dateSpy.mockRestore()
    })
  })
})
