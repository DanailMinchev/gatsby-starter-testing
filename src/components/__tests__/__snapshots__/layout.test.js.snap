// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Snapshots matches snapshot 1`] = `
<div>
  <header
    style="background: rebeccapurple; margin-bottom: 1.45rem;"
  >
    <div
      style="margin: 0px auto; max-width: 960px; padding: 1.45rem 1.0875rem;"
    >
      <h1
        style="margin: 0px;"
      >
        <a
          href="/"
          style="color: white; text-decoration: none;"
        >
          Gatsby Testing Starter
        </a>
      </h1>
    </div>
  </header>
  <div
    style="margin: 0px auto; max-width: 960px; padding: 0px 1.0875rem 1.45rem;"
  >
    <main>
      <main>
        <h1>
          Hello World
        </h1>
      </main>
    </main>
    <footer>
      © 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>
</div>
`;

exports[`PureLayout Snapshots matches snapshot 1`] = `
<div>
  <header
    style="background: rebeccapurple; margin-bottom: 1.45rem;"
  >
    <div
      style="margin: 0px auto; max-width: 960px; padding: 1.45rem 1.0875rem;"
    >
      <h1
        style="margin: 0px;"
      >
        <a
          href="/"
          style="color: white; text-decoration: none;"
        >
          Gatsby Testing Starter
        </a>
      </h1>
    </div>
  </header>
  <div
    style="margin: 0px auto; max-width: 960px; padding: 0px 1.0875rem 1.45rem;"
  >
    <main>
      <main>
        <h1>
          Hello World
        </h1>
      </main>
    </main>
    <footer>
      © 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>
</div>
`;
